core/geometry/src/curve/CurveChainWithDistanceIndex.ts

```diff
@@ -786,9 +779,8 @@ export class CurveChainWithDistanceIndex extends CurvePrimitive {
           break;
         CurveChainWithDistanceIndex._numTested++;
         const child = sortedFragment.childCurve;
-        detailA = child.closestPoint(
-          spacePoint, sortedFragment === fragment0 ? extend0 : sortedFragment === fragment1 ? extend1 : false, detailA,
-        );
```

Comment:
@dassaf4 I realized it had a bug. Fragments are not curve childs. For example my test curve had 3 child but 15 fragments. Because of that `extend` was not working fine so had to compare make below change to fix it.